presentation/src/main/java/me/androidbox/presentation/task/viewmodel/TaskDetailViewModel.kt

```diff
@@ -32,9 +100,90 @@ class TaskDetailViewModel @Inject constructor(
                     )
                 }
             }
-            is TaskDetailScreenEvent.OnShowDeleteEventAlertDialog -> TODO()
-            TaskDetailScreenEvent.OnSaveTaskDetails -> TODO()
-            is TaskDetailScreenEvent.OnShowAlarmReminderDropdown -> TODO()
+            is TaskDetailScreenEvent.OnShowDeleteEventAlertDialog -> {
+                _taskDetailScreenState.update { taskDetailScreenState ->
+                    taskDetailScreenState.copy(
+                        shouldShowDeleteAlertDialog = taskDetailScreenEvent.shouldShowDeleteAlertDialog
+                    )
+                }
+            }
+            is TaskDetailScreenEvent.OnSaveTaskDetails -> {
+                insertTaskDetails(taskDetailScreenEvent.taskId)
+            }
+            is TaskDetailScreenEvent.OnShowAlarmReminderDropdown -> {
+                _taskDetailScreenState.update { taskDetailScreenState ->
+                    taskDetailScreenState.copy(
+                        shouldOpenDropdown = taskDetailScreenEvent.shouldOpen
+                    )
+                }
+            }
+        }
+    }
+
+    private fun insertTaskDetails(taskId: String) {
+        val startDateTime = AlarmReminderProvider.getCombinedFromDate(taskDetailScreenState.value.from)
+        val remindAt = AlarmReminderProvider.getRemindAt(taskDetailScreenState.value.alarmReminderItem, startDateTime)
+
+        val task = Task(
+            id = taskId,
+            title = taskDetailScreenState.value.taskTitle,
+            description = taskDetailScreenState.value.taskDescription,
+            startDateTime = taskDetailScreenState.value.from.toEpochSecond(),
+            remindAt = remindAt.toEpochSecond(),
+            isDone = taskDetailScreenState.value.isDone)
+
+        viewModelScope.launch {
+            taskRepositoryImp.insertTask(task)
+
+            val alarmItem = task.toAlarmItem()
+            alarmScheduler.scheduleAlarmReminder(alarmItem)
+
+            _taskDetailScreenState.update { taskDetailScreenState ->
+                taskDetailScreenState.copy(
+                    isSaved = true
+                )
+            }
+
+            when(taskRepositoryImp.uploadTask(task)) {
+                ResponseState.Loading -> TODO()
+                is ResponseState.Success -> Unit
+                is ResponseState.Failure -> {
+                    /** TODO We could display a snackbar or a toast message */
+                    taskRepositoryImp.insertSyncTask(taskId)
```

Comment:
If task fails create on the /task endpoint. We add this to the syncTask table?

Reply:
Exactly

As mentioned above, all this logic could happen on the repository

Reply:
@MKiperszmid Updated, moved logic to the repository